main.go

```diff
@@ -64,8 +62,13 @@ func buildRegistryCommand(cmdName string, registry Registry, servicesConfig Serv
 		Short: "Build, tag and push an image",
 		Run: func(cmd *cobra.Command, args []string) {
 			authOption := getRegistryAuthOption(registry)
+			maxGoRoutines, err := cmd.Flags().GetInt("max-go-routines")
+			if err != nil {
+				finishWithError("failed getting max-go-routines flag", err)
+			}
 
 			var g errgroup.Group
+			g.SetLimit(maxGoRoutines)
```

Comment:
I don't really like using errgroup.
I like the idea of adding error handling to wait groups, but it is an incomplete one - you can only receive the first error that occurs